### PR TITLE
HACKING.md: Fix a few details

### PR DESCRIPTION
**Pull request by valentindavid**
There was a double `grep`. Also `ubuntu-core-initramfs` has to run as root to write in `/boot`. `create-initrd` has to be run before `create-efi`.

Also I repeated the way to repack the kernel snap instead of guessing the commands from other sections.

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -249,12 +249,21 @@ also possible to force the build of `kernel.efi` or just the initrd
...
 
 (the initrd image will also be created in the `/boot` folder). Note
 again that for RPi we need only the initrd image file.
 
+The kernel snap can be repacked with the built `kernel.efi` with
+
+```
+$ snap download --channel=20/stable pc-kernel
+$ unsquashfs -d pc-kernel pc-kernel_*.snap
+$ sudo install -m 0600 -o $(id -u) -g $(id -g) /boot/kernel.efi-$kernelver pc-kernel/kernel.efi
```
[the remark this answers is not in the document]
Comment (alfonsosanchezbeato) [reply HACKING.md:264]: I think that the purpose here is to be able to copy `/boot/kernel.efi-$kernelver`, that is root read-only, while changing the owner to your user inside the snap (@valentindavid might correct me here if I'm wrong), so the `id` commands actually are evaluated at the right moment for this.

[the remark this answers is not in the document]
Comment (valentindavid) [reply HACKING.md:264]: Correct. The unsquashed directory is owned by the user. I want to access `/boot` but which is not user accessible. I do not want the copied file to be owned by root.

File: HACKING.md [hunk 2]
```diff
@@ -249,12 +249,21 @@ also possible to force the build of `kernel.efi` or just the initrd
 respectively, with
 
 ```
-$ ubuntu-core-initramfs create-efi --kernelver=$kernelver
-$ ubuntu-core-initramfs create-initrd --kernelver=$kernelver
+$ sudo ubuntu-core-initramfs create-initrd --kernelver=$kernelver
+$ sudo ubuntu-core-initramfs create-efi --kernelver=$kernelver
```
[the remark this answers is not in the document]
Comment (valentindavid) [reply HACKING.md:253]: You probably meant `--initrd $PWD/new-initrd`.

[the remark this answers is not in the document]
Comment (bboozzoo) [reply HACKING.md:253]: Yeah, silly typo :)

[the remark this answers is not in the document]
Comment (valentindavid) [reply HACKING.md:253]: So the problem is that `HACKING.md`also instructs to `apt install linux-image-$kernelver`. And this is also not nice. So if we want to change these lines, I think we have to change the previous instructions too.

[the remark this answers is not in the document]
Comment (valentindavid) [reply HACKING.md]: I have rewritten the section.

File: HACKING.md [hunk 3]
```diff
@@ -225,36 +225,38 @@ Then, install the package in the container:
...
 We can inject it in the kernel snap as in previous sections. It is
-also possible to force the build of `kernel.efi` or just the initrd
-respectively, with
+also possible to force the build of `kernel.efi` (or just the initrd), with:
 
 ```
-$ ubuntu-core-initramfs create-efi --kernelver=$kernelver
-$ ubuntu-core-initramfs create-initrd --kernelver=$kernelver
+$ ubuntu-core-initramfs create-initrd --kernelver=$kernelver --kerneldir pc-kernel/modules/${kernelver} --firmwaredir pc-kernel/firmware --output initrd.img
+$ ubuntu-core-initramfs create-efi --kernelver=$kernelver --initrd initrd.img --kernel kernel.img --output kernel.efi
```
Comment (alfonsosanchezbeato) [inline HACKING.md:249]: Should this be `--kernel kernel.img-"${kernelver}"`?

File: HACKING.md [hunk 4]
```diff
@@ -225,36 +225,38 @@ Then, install the package in the container:
...
 ```
 
-(the initrd image will also be created in the `/boot` folder). Note
-again that for RPi we need only the initrd image file.
+Note that for RPi we need only the initrd image file.
+
+Now `kernel.efi-5.4.0-87-generic` has been created with the new initramfs. We can put it back into the snap:
+
+```
+$ cp kernel.efi-$kernelver pc-kernel/kernel.efi
```
Comment (alfonsosanchezbeato) [inline HACKING.md:257]: And this one `cp kernel.efi pc-kernel/kernel.efi`, unless you change the previous command too.

**Review by alfonsosanchezbeato — COMMENTED** [review]
Looks good, just have a couple of minor nits. Also, you should change the phrase that says "As we are going to install a kernel, it is preferable to run these instructions inside an lxd container or VM" a bit up in the document.

File: HACKING.md (hunk 3 above)
Comment (valentindavid) [reply HACKING.md:249]: No. The kernelver is automatically appended by `ubuntu-core-initramfs`.

**Comment by valentindavid** [comment]
> Also, you should change the phrase that says "As we are going to install a kernel, it is preferable to run these instructions inside an lxd container or VM" a bit up in the document.

Right. I have now removed this sentence.

**Review by alfonsosanchezbeato — APPROVED** [review]
Cool, thanks for the changes

**Review by anonymouse64 — APPROVED** [review]
lgtm

**Review by mardy — APPROVED** [review]
LGTM!

**Pull request merged**